File: src/380.go
type RandomizedSet struct {
	mp map[int]bool
}

func Constructor() RandomizedSet {
	return RandomizedSet{
		mp: make(map[int]bool),
	}
}

func (this *RandomizedSet) Insert(val int) bool {
	if _, ok := this.mp[val]; !ok {
		this.mp[val] = true
		return true
	}
	return false
}

func (this *RandomizedSet) Remove(val int) bool {
	if _, ok := this.mp[val]; ok {
		delete(this.mp, val)
		return true
	}
	return false
}

func (this *RandomizedSet) GetRandom() int {
	l := len(this.mp)
	randNumber := rand.Intn(l)
	counter := 0

	for k := range this.mp {
		if counter == randNumber {
			return k
		}
		counter++
	}
	return -1
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Insert(val);
 * param_2 := obj.Remove(val);
 * param_3 := obj.GetRandom();
 */

File: src/380.py
from typing import Set


class RandomizedSet:
    def __init__(self):
        self.s = set()

    def insert(self, val: int) -> bool:
        self.s.add(val)
        return True

    def remove(self, val: int) -> bool:
        try:
            self.s.remove(val)
            return True
        except KeyError:
            return False

    def getRandom(self) -> int:
        self.s.pop()


# Your RandomizedSet object will be instantiated and called as such:
obj = RandomizedSet()
obj.insert(1)
obj.remove(2)
obj.insert(2)
obj.getRandom()
obj.remove(1)
obj.insert(2)
obj.getRandom()
